elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/schema/SQLSchema.java

```diff
@@ -138,14 +141,36 @@ public static String getColumnName(EntityDictionary entityDictionary, Class<?> c
         }
     }
 
+    /**
+     * Maps an entity class to a physical table of subselect query, if neither {@link Table} nor {@link Subselect}
+     * annotation is present on this class, use the class alias as default.
+     *
+     * @param entityDictionary The dictionary for this elide instance.
+     * @param cls The entity class.
+     * @return The physical SQL table or subselect query.
+     */
+    public static String getTableOrSubselect(EntityDictionary entityDictionary, Class<?> cls) {
+        Subselect subselectAnnotation = entityDictionary.getAnnotation(cls, Subselect.class);
+
+        if (subselectAnnotation == null) {
+            Table tableAnnotation = entityDictionary.getAnnotation(cls, Table.class);
+
+            return (tableAnnotation == null)
+                    ? entityDictionary.getJsonAliasFor(cls)
+                    : tableAnnotation.name();
+        } else {
+            return "(" + subselectAnnotation.value() + ")";
```

Comment:
I feel a bit uncomfortable adding SQL formatting here like parenthesis.  That's spreading the responsibility of generating the SQL code across many classes.  The schema should just be about returning metadata - not doing SQL expansion.  

Also - should this really be a static method?   @hellohanchen - I think this overlaps with your metadata work.  We need a metadata repository about AnalyticViews and Entities.  An Entity should be able to fetch where its table name comes from.  I think your PR is going to be a pretty large refactor of where this kind of logic lives.